build-tools/install-additional-software.sh

```diff
@@ -0,0 +1,145 @@
+#!/bin/sh
+
+#
+# Copyright 2019-2020 JetBrains s.r.o.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# Uncomment this in case uou need extra debug info
+#set -e # Any command which returns non-zero exit code will cause this shell script to exit immediately
+#set -x # Activate debugging to show execution details: all commands will be printed before execution
+
+DOWNLOAD_URL=$1
+
+# In SOFTWARE_LIST block add software package names. This string was directly passed to `apt install` command
+# In ADDITIONAL_COMMANDS you can add any BASH commands. Please, note that you need to escape $ character by backslash (\)
+
+# Detect type of IDE we use and create a list of required software
+case $DOWNLOAD_URL in
+
+#
+# Block for CLion
+  *CLion*)
+    SOFTWARE_LIST=$(sed -z 's/  */ /g;s/\n//g' <<- EOM
```

Comment:
Interesting script. However, there is a lot of duplicated code. Can we somehow extract duplicated code to a function or to a constant? I mean this part: `sed -z 's/ */ /g;s/\n//g'` or maybe even including `EOM`...

Reply:
I also think about it. But this is just a PoC and I dont be sure that my idea will be approved. I can move `sed ...` block to constant, but `EOM` is a part of heredoc string definition, so I cant move it